cmd/gapir/cc/main.cpp

```diff
@@ -132,25 +129,24 @@ void android_main(struct android_app* app) {
     CrashHandler crashHandler;
 
     const char* pipe = pipeName();
-    auto conn = SocketConnection::createPipe(pipe, "");
-    if (conn == nullptr) {
-        GAPID_FATAL("Failed to create abstract local port: %s", pipe);
-    }
+    std::string internal_data_path = std::string(app->activity->internalDataPath);
+    std::string socket_file_path = internal_data_path + "/" + std::string(pipe);
+    std::string uri = std::string("unix://") + socket_file_path;
 
     __android_log_print(ANDROID_LOG_DEBUG, "GAPIR",
             "Started Graphics API Replay daemon.\n"
-            "Listening on localabstract port '%s'\n"
+            "Listening on unix socket '%s'\n"
             "Supported ABIs: %s\n",
-            pipe, core::supportedABIs());
-
-    // Note if you want to create a disk cache create it under:
-    // app->activity->internalDataPath
-    std::thread listening_thread([&]() {
-      listenConnections(conn.get(), nullptr, nullptr, Connection::NO_TIMEOUT,
-                        &memoryManager, crashHandler);
-    });
-    bool alive = true;
-    while (alive) {
+            uri.c_str(), core::supportedABIs());
+
+    std::unique_ptr<Server> server = Setup(
+        uri.c_str(), nullptr, nullptr, 0 /*no timeout?*/, &crashHandler, &memoryManager);
+    std::thread waiting_thread([&]() { server.get()->wait(); });
+    if (chmod(socket_file_path.c_str(), S_IRUSR|S_IWUSR|S_IXUSR|S_IROTH|S_IWOTH|S_IXOTH)) {
```

Comment:
Do we really need executable flags here? That seems overly-broad.

Reply:
This definitely should be a TODO